source/grapher.js

```diff
@@ -737,11 +737,53 @@ grapher.Edge = class {
         const createElement = (name) => {
             return document.createElementNS('http://www.w3.org/2000/svg', name);
         };
+        const getDatatype = (from) => {
+            if (from && from.value) {
+                const from_value = from.value;
+                let from_value_src;
+                if (from_value.outputs && from_value.outputs[0].value) {
```

Comment:
```suggestion
                if (from_value.outputs && from_value.outputs[0]?.value) {
```

Reply:
Currently not use optional chaining. Check browser support matrix and apply consistently across the codebase.